compiler/lib/effects.ml

```diff
@@ -557,9 +655,33 @@ let cps_block ~st ~k pc block =
         Some
           (fun ~k ->
             let exact =
-              exact || Global_flow.exact_call st.flow_info f (List.length args)
+              exact
+              (* If this function is unknown to the global flow analysis, then it was
+                 introduced by the lambda lifting and is exact *)
+              || Var.idx f >= Var.Tbl.length st.flow_info.info_approximation
+              || Global_flow.exact_call st.flow_info f (List.length args)
             in
             tail_call ~st ~exact ~in_cps:true ~check:true ~f (args @ [ k ]))
+    | Prim (Extern "caml_assume_no_perform", [ Pv f ])
+      when (not (double_translate ())) && Var.Set.mem x st.cps_needed ->
+        (* Translated like the [Apply] case, with a unit argument *)
+        Some
+          (fun ~k ->
+            let exact =
+              (* If this function is unknown to the global flow analysis, then it was
+                 introduced by the lambda lifting and is exact *)
+              Var.idx f >= Var.Tbl.length st.flow_info.info_approximation
+              || Global_flow.exact_call st.flow_info f 1
```

Comment:
Same comment as above: if we don't know the function, we must assume the call is not exact.